src/features/create-dao/steps/DetailsForm.test.tsx

```diff
@@ -0,0 +1,148 @@
+//- React Imports
+import React from 'react';
+import { render, waitFor, screen, fireEvent } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+//- Component Imports
+import { DetailsForm, DetailsFormProps } from './DetailsForm';
+import { ZUIProvider } from '@zero-tech/zui/ZUIProvider';
+
+//- Type Imports
+import { CreateDAOFormContext } from '../CreateDAOFormContext';
+
+let onSubmit = jest.fn();
+
+const DEFAULT_PROPS: DetailsFormProps = {
+	onClose: jest.fn(),
+};
+
+const DEFAULT_PROVIDER_VALUES = {
+	stepId: 'details',
+	title: 'Create DAO',
+	details: {
+		mediaType: undefined,
+		previewUrl: '',
+		name: '',
+		znaAddress: '',
+		description: '',
+	},
+	governance: {
+		votingProcess: '',
+		votingPeriod: '',
+		votingSystem: '',
+		daoTokenAddress: '',
+		votingThreshold: '',
+	},
+	treasury: {
+		gnosisSafe: '',
+	},
+	onStepUpdate: jest.fn(),
+	onTitleUpdate: jest.fn(),
+	onDetailsSubmit: onSubmit,
+	onGovernanceSubmit: jest.fn(),
+	onTreasurySubmit: jest.fn(),
+	onLaunchSubmit: jest.fn(),
+};
+
+const renderDetailsForm = (
```

Comment:
Hey Ash, could we add the render to each `test` block rather than create a function here for it. Brett and I ironed out some practices - Here are a couple of test files from zUI for reference

https://github.com/zer0-os/zUI/blob/30df63a8ffdff9f8bc7401d13bc0a41a0930fc63/src/components/Card/Card.test.tsx

https://github.com/zer0-os/zUI/blob/30df63a8ffdff9f8bc7401d13bc0a41a0930fc63/src/components/GridCard/GridCard.test.tsx

https://github.com/zer0-os/zUI/blob/30df63a8ffdff9f8bc7401d13bc0a41a0930fc63/src/components/TextStack/TextStack.test.tsx

Take a look at how we're testing classNames and forwarding props :) Testing will follow these practices closely as we move forward. Thanks mate!